armi/settings/fwSettings/tightCouplingSettings.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2019 TerraPower, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The data structures and schema of the tight coupling settings.
+
+These are advanced/compound settings that are carried along in the normal cs
+object but aren't simple key/value pairs.
+"""
+
+from typing import Dict, Union
+
+import voluptuous as vol
+
+from armi.settings import Setting
+
+_SCHEMA = vol.Schema(
+    {
+        str: vol.Schema(
+            {
+                vol.Required("parameter"): str,
+                vol.Required("convergence"): vol.Coerce(float),
+            }
+        )
+    }
+)
+
+
+class TightCouplingSettings(dict):
+    """
+    Dictionary with keys of Interface functions and a dictionary value.
+
+    Notes
+    -----
+    The dictionary value for each Interface function is required to contain a ``parameter``
+    and a ``convergence`` key with string and float values, respectively. No other
+    keys are allowed.
+
+    Examples
+    --------
+    couplingSettings = TightCouplingSettings({'globalFlux': {'parameter': 'keff', 'convergence': 1e-05}})
+    """
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__} with Interface functions {self.keys()}>"
+
+
+def serializeTightCouplingSettings(
+    tightCouplingSettingsDict: Union[TightCouplingSettings, Dict]
+) -> Dict[str, Dict]:
+    """
+    Return a serialized form of the ``TightCouplingSettings`` as a dictionary.
+
+    Notes
+    -----
+    Attributes that are not set (i.e., set to None) will be skipped.
+    """
+    if not isinstance(tightCouplingSettingsDict, dict):
+        raise TypeError(f"Expected a dictionary for {tightCouplingSettingsDict}")
```

Comment:
The unit tests within `test_tightCouplingSettings.py` have great coverage for everything except this TypeError. Can we either 1) add a test so this is covered, or 2) remove this exception?

Reply:
Sure, please add one!

Reply:
https://github.com/terrapower/armi/pull/1033/commits/e279efcc83254284cc7f50bd7c2ca79264519b67